### PR TITLE
fix(ui-library): align hint and error message icons

### LINKED ISSUE #515: Align how Hint & Error Messages are displayed in our component examples
#### Description / User Story

Currently we are not displaying Error Messages consistently in design. Sometimes they have an Icon, sometimes not. We should align this for all components.

In code we neither show an ErrorMessage nor an ErrorMessageIcon per default.

---

Proposal how it should be in Code & in Figma:
- (blrInfo) This is a small hint
- (blrErrorFilled) OMG it's an error

Misalignments in Code to fix, if following the proposal:
- Select: Change blrInfo to blrErrorFilled in Error message in Form Caption Group example
- Radio: Change blr360 to blrInfo in Hint message in Form Caption Group example
- RadioGroup: blrErrorFilled Icon is missing in Error message in Form Caption Group example

Misalignments in Figma to fix, if following the proposal:
- Radio: Add the HintMessage Icon
- TextInput TypeText: Add HasCaption sticker with and without all captions in rest and in error state (similar to NumberInput) > this may result also in fixing the Error and Hint Message to not being bold

----
#### Acceptance Criteria
- [x] All components in Figma are aligned in the way we display Error and Hint Messages
- [x] All components in Storybook are aligned in the way we display Error and Hint Messages
   - [x] Select: blrInfo changed to blrErrorFilled in Error message in Form Caption Group example of the Select component
   - [x] Radio: blr360 changed to blrInfo in Hint message in Form Caption Group example of the Radio Component
   - [x] Radio Group: blrErrorFilled Icon has been added in Error message in Form Caption Group example of the Radio Group component
   - [ ] TextInput: default error message icon is "undefined" in the Text Input
   - [x] TextArea: default error message icon is "undefined" in the Text Area
   - [x] Select: default error message icon is "undefined" in the Select
- [x] A unit test has been implemented to make sure the error message icon is always undefined per default in Select, Radio, Radio Group, TextInput, Text Area and Checkbox

----
#### Background information
- ...

### PR DESCRIPTION
**Pull request by RubirajAccenture**
closes #515

### CODE REVIEW

**Review by davidken91 — APPROVED** [review]

File: packages/ui-library/src/components/checkbox/index.test.ts [hunk 1]
```diff
@@ -34,6 +34,23 @@ describe('blr-checkbox', () => {
     expect(className).to.contain('input-control');
   });
 
+  it('has error Icon is set to undefined', async () => {
```
Comment (angsherpa456) [inline packages/ui-library/src/components/checkbox/index.test.ts:37]: Please adjust it to it('has error Icon set to undefined'...) [remove 'is' as we discussed yesterday]

File: packages/ui-library/src/components/input-field-text/index.test.ts [hunk 2]
```diff
@@ -121,6 +121,23 @@ describe('blr-input-field-text', () => {
...
+        ...sampleParams,
+        hasHint: false,
+        hasError: true,
+        errorMessage: 'error',
+        errorMessageIcon: undefined,
+      })
+    );
+
+    const formCaption = querySelectorDeep('.blr-form-caption', element?.getRootNode() as HTMLElement);
+
```
Comment (angsherpa456) [inline packages/ui-library/src/components/input-field-text/index.test.ts:136]: Please remove this extra new line.

File: packages/ui-library/src/components/radio-group/index.test.ts [hunk 3]
```diff
@@ -81,4 +81,21 @@ describe('blr-radio-group', () => {
 
     expect(className).not.to.contain('error');
   });
+
+  it('has error Icon set to undefined', async () => {
+    const element = await fixture(
+      BlrRadioGroupRenderFunction({
+        ...sampleParams,
+        hasHint: false,
+        hasError: false,
```
Comment (angsherpa456) [inline packages/ui-library/src/components/radio-group/index.test.ts:90]: hasError needs to be true

File: packages/ui-library/src/components/radio/index.test.ts [hunk 4]
```diff
@@ -99,4 +100,21 @@ describe('blr-radio', () => {
 
     expect(className).not.to.contain('error');
   });
+
+  it('has error Icon set to undefined', async () => {
+    const element = await fixture(
+      BlrRadioRenderFunction({
+        ...sampleParams,
+        hasHint: false,
+        hasError: false,
```
Comment (angsherpa456) [inline packages/ui-library/src/components/radio/index.test.ts:109]: hasError needs to be true

File: packages/ui-library/src/components/radio/index.test.ts [hunk 5]
```diff
@@ -99,4 +100,21 @@ describe('blr-radio', () => {
...
+        errorIcon: undefined,
+      })
+    );
+    const labelWrapper = querySelectorDeep('.label-wrapper', element.getRootNode() as HTMLElement);
+    const captionWrapper = querySelectorDeep('.caption-wraper', labelWrapper?.getRootNode() as HTMLElement);
+    const formCaption = querySelectorDeep('.blr-form-caption', captionWrapper?.getRootNode() as HTMLElement);
+
```
Comment (angsherpa456) [inline packages/ui-library/src/components/radio/index.test.ts:116]: This extra new line can be removed for the consistency purpose. Right now they are mixed up.

File: packages/ui-library/src/components/textarea/index.test.ts [hunk 6]
```diff
@@ -118,6 +118,23 @@ describe('blr-textarea', () => {
...
+        ...sampleParams,
+        hasHint: false,
+        hasError: false,
+        errorMessageIcon: undefined,
+      })
+    );
+
+    const textarea = querySelectorDeep('textarea', element.getRootNode() as HTMLElement);
+    const formCaption = querySelectorDeep('.blr-form-caption', textarea?.getRootNode() as HTMLElement);
+
```
Comment (angsherpa456) [inline packages/ui-library/src/components/textarea/index.test.ts:133]: Please remove this extra new line.

**Review by thrbnhrtmnn — CHANGES_REQUESTED** [review]
Hey @RubirajAccenture , I just found one small thing. If I set the Input Field Text component to "hasError"=true, I see a blrInfo icon. This should not be there, the default icon should be undefined. Please also check that the test that you implemented also works and finds this issue.

**Review by angsherpa456 — APPROVED** [review]
LGTM

**Review by davidken91 — APPROVED** [review]

**Review by thrbnhrtmnn — APPROVED** [review]
Approved

**Pull request merged**